students/saleta_sebastian/lesson_12_debugging/validated_user_base.py

```diff
@@ -0,0 +1,130 @@
+import argparse
+import csv
+import logging
+import os
+import shutil
+import sys
+
+from lesson_08_regular_expressions import \
+    strong_password_detection, email_validator, phone_number_validator,\
+    postal_code_validator
+
+
+sys.path.append('..')
+logging.basicConfig(level=logging.DEBUG, format=(' %(asctime)s - %(levelname)s'
+                                                 ' - %(message)s'))
+
+TEST_FILES_DIRECTORY = 'test_files'
+TEST_FILE_NAME = 'user_base.csv'
+TEMP_FILE_NAME = 'temp_base.csv'
+COLUMN_TITLE_ROW = "email, password, phone number, postal code \n"
+
+logger = logging.getLogger('__name__')
+
+
+def set_logger_level(lvl):
+    if lvl == 0:
+        logging.disable(logging.CRITICAL)
+    elif lvl == 1:
+        logger.setLevel('WARNING')
+    elif lvl == 2:
+        logger.setLevel('INFO')
+
+
+def check_arg(args=None):
+    parser = argparse.ArgumentParser(description='Script to validate and save'
+                                                 ' user data in csv file')
+    parser.add_argument('-v', '--verbose',
+                        help=('Verbose level.'
+                              '0 - disabled, '
+                              '1 - warnings and errors enabled '
+                              '2 - info enabled'),
+                        type=int)
+    results = parser.parse_args(args)
+    return results.verbose
+
+
+def validate_user_data(email, password, phone_number, postal_code):
+    logging.info('Validating user data')
+    if not email_validator.check_email(email):
+        logging.info('Email adress is invalid')
+        raise Exception('Email address is invalid')
```

Comment:
You can consider implementing your own exception (before line 25), have a look at this:
```
class ValidateUserBaseException(Exception):
    module = 'ValidateUserBaseException'

    def __init__(self, message):
        super(ValidateUserBaseException, self).__init__(self.module + ': ' + message)
```
and then
```raise ValidateUserBaseException('Email address is invalid')```
Of course it can stay as it's now